core/agent/agent_executor.py

```
@@ -108,15 +109,28 @@ async def run(
+            # Try models in order, failover to next on exception
+            llm_resp = None
+            for model_idx, model in enumerate(model_group):
+                try:
+                    llm_resp = await model.chat(request)
+                    llm_model = model
+                    if model_idx > 0:
+                        provider_name = llm_model.model.provider_name
+                        model_id = model.model.model_id
+                        llm_logger.info(f"[{sid}] Successfully switched to model: {model_id} ({provider_name})")
+                    break
+                except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)